pkg/web/apps/apps_http_handler_test.go

```diff
@@ -88,3 +105,60 @@ func Test_HttpHandler_UpdateApp(t *testing.T) {
 		//assert.Equal(t, string(appJson)+"\n", rec.Body.String())
 	}
 }
+
+func Test_httpHandler_GetAppByID(t *testing.T) {
+	config := config.Get()
+  APIRoutePrefix := config.APIRoutePrefix
+	// make and configure a mocked Service
+	app := helpers.GetMockApp()
+	mockedAppService := &ServiceMock{
+		GetAppByIDFunc: func(ID string) (*models.App, error) {
+			if app != nil { 
+				return app, nil
+			}
+			return nil, models.ErrNotFound
+		},
+	}
+	// Setup
+	e := echo.New()
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	c.SetPath(APIRoutePrefix + "/apps/:id")
+	c.SetParamNames("id")
+	h := NewHTTPHandler(e, mockedAppService)
+	type fields struct {
+		Service Service
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		id      string
+		wantErr bool
+		want    *models.App
+	}{
+		{
+			name: "Get app by id should return an app",
+			id: app.ID,
+			want: app,
+		},
+		{
+			name: "Get app by id using an invalid id format should return an error",
+			id: "some string that should fail",
+			want: nil,
+		},
+		{
+			name:"Get app by id valid format but id not found",
+			id: "cb1becdd-7726-4902-9014-6fb2296b9ae6",
+			want: nil,
+		},
+	}
+	for _, tt := range tests {
+		c.SetParamValues(tt.id)
+		t.Run(tt.name, func(t *testing.T) {
+			if err := h.GetAppByID(c); (err != nil) != tt.wantErr {
+				t.Errorf("httpHandler.GetAppByID() error = %v, wantErr %v", err, tt.wantErr)
```

Comment:
It is just checking if has an error or not, when we should check the HTTP code status and if the JSON returned by the handler is == what is expected.

Reply:
Hi @craicoverflow, 
How we do with the testing the previous validation here as the follows example?

```
// Assertions
	if assert.NoError(t, h.UpdateAppVersions(c)) {
		assert.Equal(t, http.StatusOK, rec.Code)
		assert.Equal(t, string(appJson)+"\n", rec.Body.String())
	}
```

Reply:
Hi @camilamacedo86 I cannot remember off the top of my head but it would not be too dissimilar to how `testify/assert` is doing it. `testify/assert` is comparing the values in `.Equal()`, the standard testing library is doing the same thing except in an _if_ statement. See the [`httperrors_test.go`](https://github.com/aerogear/mobile-security-service/blob/master/pkg/httperrors/httperrors_test.go#L171) file for how it does it.

See below for something similar to what you have. **Note**: this is not tested.

```go
// set up mock context
e := echo.New()
req := httptest.NewRequest(http.MethodPut, "/", nil)
rec := httptest.NewRecorder()
c := e.NewContext(req, rec)

c.SetPath("/api/apps")
h := NewHTTPHandler(e, mockService)

err := h.UpdateAppVersions(c)

if err != nil && !tt.wantErr {
    t.Errorf("HTTPHandler.UpdateAppVersions() err = %v, wantErr = %v", err, tt.wantErr)
}

if rec.Code != http.StatusOK {
    t.Errorf("HTTPHandler.UpdateAppVersions() statusCode = %v, wantCode = %v", rec.Code, http.StatusOK)
}

expected := `some-json`

if resBody != expected {
	t.Errorf("HTTPHandler.UpdateAppVersions() want = %v, wantCode = %v", expected, resBody)
}
```

Reply:
I did the tests that are missing here  https://github.com/aerogear/mobile-security-service/pull/72/files. 
@austincunningham I hope that it can be helpful.

Reply:
:+1:  @camilamacedo86  very helpful thanks